### PR TITLE
Roma class-1 week-1 modification

### PR DESCRIPTION
**Pull request by milleymilli**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,5 +15,8 @@
   "bugs": {
     "url": "https://github.com/CodeYourFuture/js-exercises/issues"
   },
-  "homepage": "https://github.com/CodeYourFuture/js-exercises#readme"
```
Comment (jslopezgithub) [inline package.json]: Test comment

File: week-1-practise/level-2/2-format-currency.js [hunk 2]
```diff
@@ -2,24 +2,21 @@
   CURRENCY FORMATTING
   ===================
   The business has informed you that prices must have 2 decimal places
-  They must also start with the currency symbol
+  They must also start with the currency symbol 
   Write a function that transforms numbers into the format £0.00
 */
 
-function formatCurrency() {}
+formatCurrency = price => "£" + price.toFixed(2);
```
Comment (milleymilli) [inline week-1-practise/level-2/2-format-currency.js]: //function formatCurrency() {}
forget to put it in back slash

File: week-1-practise/level-2/2-format-currency.js [hunk 3]
```diff
@@ -2,24 +2,21 @@
...
 
 /* DO NOT EDIT BELOW THIS LINE */
 /* --------------------------- */
-
-function calculateSalesTax(price) {
-  return price * 0.2;
-}
+calculateSalesTax = price => price * 0.2;
```
Comment (milleymilli) [inline week-1-practise/level-2/2-format-currency.js]: line 13 assume it it's inside a comment
//calculateSalesTax = price => price * 0.2;

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 4]
```diff
@@ -7,17 +7,28 @@
   Find a way to add 1% to all currency conversions (think about the DRY principle)
 */
 
-function convertToUSD(price) {}
-function convertToBRL(price) {}
-
+function convertToUSD(price) {
+  return price * 1.4 + 0.01 * price;
```
Comment (jslopezgithub) [inline week-1-practise/level-2/4-currency-conversion-2.js]: You could have written another function to simplify your coding.

```suggestion
function convertToUSD(price) {
  return convert(price, 1.4);
}

function convertToBRL(price) {
  return convert(price, 4.91);
}

function convert(price, exchangeRate) {
  return price * exchangeRate * 0.99;
}
```

File: week-2-practise/level-1/5-indexOf.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 var itemToFind = 7;
 var arr = [3, 5, 61, 7, 123];
-var index; // ONLY EDIT THIS LINE
+var index = arr.indexOf("h"); // ONLY EDIT THIS LINE
```
Comment (jslopezgithub) [inline week-2-practise/level-1/5-indexOf.js]: should be
```suggestion
var index = arr.indexOf(itemToFind); // ONLY EDIT THIS LINE
```

File: week-2-practise/level-2/2-validation.js [hunk 6]
```diff
@@ -4,7 +4,17 @@
 // - is less than or equal to 100
 // Tip: write other small functions for each requirement
 
-function validate(num) {}
+function validate(num) {
```
Comment (jslopezgithub) [inline week-2-practise/level-2/2-validation.js]:
```suggestion
  return typeof num === "number" && num > 0 && num <= 100;
```

File: week-2-practise/level-3/2-validation.js [hunk 7]
```diff
@@ -5,8 +5,21 @@
 // Tip: write other small functions for each requirement
 
 var excludedNums = [6, 14, 91, 111];
-
-function validate(num) {}
+function validate(num) {
```
Comment (jslopezgithub) [inline week-2-practise/level-3/2-validation.js]:
```suggestion
  return (typeof num === "number" && Number.isInteger(num) && !excludedNums.includes(num));
```

File: week-2/D-predicates/exercise.js [hunk 8]
```diff
@@ -7,12 +7,20 @@
 
 // Finish the predicate function to test if the passed number is negative (less than zero)
 function isNegative(number) {
-
+  if (number < 0) {
```
Comment (jslopezgithub) [inline week-2/D-predicates/exercise.js]:
```suggestion
  return number < 0
```

File: week-2/D-predicates/exercise.js [hunk 9]
```diff
@@ -7,12 +7,20 @@
...
+    return true;
+  } else {
+    return false;
+  }
 }
 
 // Finish the predicate function to test if the passed number is between 0 and 10
 function isBetweenZeroAnd10(number) {
-
+  if (number > 0 && number < 10) {
```
Comment (jslopezgithub) [inline week-2/D-predicates/exercise.js]:
```suggestion
  return number > 0 && number < 11;
```

File: week-2/F-logical-operators/exercise.js [hunk 10]
```diff
@@ -11,14 +11,16 @@ var cssLevel = 4;
 
 // Finish the statement to check whether HTML, CSS knowledge are above 5
 // (hint: use the comparison operator from before)
-var htmlLevelAbove5;
-var cssLevelAbove5;
+var htmlLevelAbove5 = htmlLevel > 5 ? true : false;
```
Comment (jslopezgithub) [inline week-2/F-logical-operators/exercise.js]:
```suggestion
var htmlLevelAbove5 = htmlLevel > 5;
var cssLevelAbove5 = cssLevel > 5;
```

File: week-2/F-logical-operators/exercise.js [hunk 11]
```diff
@@ -11,14 +11,16 @@ var cssLevel = 4;
...
+var cssLevelAbove5 = cssLevel > 4 ? true : false;
 
 // Finish the next two statement
 // Use the previous variables and logical operators
 // Do not "hardcode" the answers
-var cssAndHtmlAbove5;
-var cssOrHtmlAbove5;
+var cssAndHtmlAbove5 =
```
Comment (jslopezgithub) [inline week-2/F-logical-operators/exercise.js]:
```suggestion
var cssAndHtmlAbove5 = cssLevelAbove5 && htmlLevelAbove5;
var cssOrHtmlAbove5 = cssLevelAbove5 || htmlLevelAbove5;
```

File: week-2/F-logical-operators/exercise2.js [hunk 12]
```diff
@@ -5,15 +5,43 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
-
+function isNegative(num) {
```
Comment (jslopezgithub) [inline week-2/F-logical-operators/exercise2.js]: function isNegative(num) {
  return num < 0;
}

function isBetween5and10(num) {
  return num >= 5 && num <= 10;
}

function isShortName(name) {
  return name.length < 10;
}

function startsWithD(name) {
  return name[0] === "D";
}

File: week-2/G-conditionals-2/exercise-1.js [hunk 13]
```diff
@@ -5,9 +5,17 @@
...
 */
-
+/*
+let negative = "is negative";
+let positive = "is positive";
+const negativeOrPositive = number => (number < 0 ? negative : positive);
+*/
 function negativeOrPositive(number) {
-
+  if (number < 0) {
```
Comment (jslopezgithub) [inline week-2/G-conditionals-2/exercise-1.js]:
```suggestion
  return number < 0 ? "negative" : "positive";
```

File: week-2/G-conditionals-2/exercise-2.js [hunk 14]
```diff
@@ -8,7 +8,11 @@
 */
 
 function studentPassed(grade) {
-
+  if (grade > 50) {
```
Comment (jslopezgithub) [inline week-2/G-conditionals-2/exercise-2.js]:
```suggestion
  return grade < 50 ? "passed" : "failed"
```

File: week-2/G-conditionals-2/exercise-4.js [hunk 15]
```diff
@@ -7,23 +7,27 @@
 
   Hint: Google how to check if a string contains a word
 */
-
-function containsCode(sentence) {
-
+let containsCode = sentence => (sentence.includes("code") ? true : false);
```
Comment (jslopezgithub) [inline week-2/G-conditionals-2/exercise-4.js]:
```suggestion
let containsCode = sentence => sentence.includes("code");
```

File: week-2/I-array-properties/exercise.js [hunk 16]
```diff
@@ -2,15 +2,17 @@
   Array properites
   ----------------
   Complete the function to test if an array is empty (has no values in it)
-
 */
 
 function isEmpty(arr) {
-  return; // complete this statement
+  if (arr.length == 0) {
```
Comment (jslopezgithub) [inline week-2/I-array-properties/exercise.js]:
```suggestion
  return arr.length == 0;
```

File: week-4/D-methods/exercise-5.js [hunk 17]
```diff
@@ -9,35 +9,49 @@ only if the inserted amount is greater or equal than the price of the coffee!
...
+    cappuccino: 2.4,
+    blackCoffee: 1.5,
+    flatWhite: 3.0
+  },
+  insertedAmount: 0,
+  insertMoney: function(amount) {
+    this.insertedAmount = amount;
+  },
+  getCoffee: function(coffee) {
+    if (coffeeMachine.prices[coffee] <= coffeeMachine.insertedAmount) {
```
Comment (jslopezgithub) [inline week-4/D-methods/exercise-5.js]: You need to subtract money from the machine.

File: week-5/B-callbacks/exercise.js [hunk 18]
```diff
@@ -14,10 +14,10 @@
   EXPECTED RESULT: The #exercise1 element has textContent = "YOUR NAME" when
   the button is clicked
 */
-document.querySelector('#button1').addEventListener('click', exercise1)
+document.querySelector("#button1").addEventListener("click", exercise1);
 
 function exercise1() {
-  // Write your implementation here
+  return (document.getElementById("exercise-1").innerHTML = Milley); // Write your implementation here
```
Comment (jslopezgithub) [inline week-5/B-callbacks/exercise.js]: no need to return anything here. Milley should be in quotes.

File: week-5/B-callbacks/exercise.js [hunk 19]
```diff
@@ -34,10 +34,10 @@ function exercise1() {
   EXPECTED RESULT: The #exercise2 element has textContent = "Hello from the
   function caller"
 */
-functionThatCallsBack(exercise2)
+functionThatCallsBack(exercise2);
 
 function exercise2(result) {
-  // Write your implementation here
+  return result; // Write your implementation here
```
Comment (jslopezgithub) [inline week-5/B-callbacks/exercise.js]: This should have updated "#exercise2"

File: week-5/C-promises/exercise.js [hunk 20]
```diff
@@ -14,20 +14,22 @@
   Promise"
 */
 function exercise1() {
-  var promise1 = resolvedPromise()
+  var promise1 = resolvedPromise();
+  promise1.then(res => console.log(res));
```
Comment (jslopezgithub) [inline week-5/C-promises/exercise.js]: Read the instructions for exercise1 again.

File: week-5/C-promises/exercise.js [hunk 21]
```diff
@@ -14,20 +14,22 @@
...
-  Insert the result of promise2 into the #exercise2 element. Hint: .catch()
+  Insert the result of jibran into the #exercise2 element. Hint: .catch()
 
   EXPECTED RESULT: The #exercise2 element has textContent = "A Unpromising
   Promise"
 */
 function exercise2() {
-  var promise2 = rejectedPromise()
+  var jibran = rejectedPromise();
+  jibran.catch(err => console.log(err));
```
Comment (jslopezgithub) [inline week-5/C-promises/exercise.js]: Read the instructions

File: week-5/C-promises/exercise.js [hunk 22]
```diff
@@ -40,7 +42,8 @@ function exercise2() {
   EXPECTED RESULT: The #exercise3 element has textContent = "A Longer Promise"
 */
 function exercise3() {
-  var promise3 = delayedPromise()
+  var promise3 = delayedPromise();
+  promise3.then(res => console.log(res));
```
Comment (jslopezgithub) [inline week-5/C-promises/exercise.js]: Read the instructions

File: week-5/F-webchat/exercise-1.js [hunk 23]
```diff
@@ -33,5 +34,11 @@ When you open index.html in your browser, it should display the existing message
 
 */
 
+// Write your code here
 
-// Write your code here
\ No newline at end of file
+fetch("https://codeyourfuture.herokuapp.com/api/messages")
+  .then(response => response.text())
```
Comment (jslopezgithub) [inline week-5/F-webchat/exercise-1.js]: should be `response.json()`

File: week-5/F-webchat/exercise-1.js [hunk 24]
```diff
@@ -33,5 +34,11 @@ When you open index.html in your browser, it should display the existing message
...
+  .then(function(mes) {
+    document.querySelector("#message-list").innerHTML = mes;
```
Comment (jslopezgithub) [inline week-5/F-webchat/exercise-1.js]: You should use messages.forEach() to append each message separately as a new node

File: week-5/F-webchat/exercise-2.js [hunk 25]
```diff
@@ -12,18 +12,36 @@ send the message to store it on the remote server. Use the following API:
...
-
-1) The input field should be empty.
-2) When you refresh the page in your browser, you should be able to see your new message in the message list.
 */
 
-
-// Write your code here
+document.getElementById("submit").addEventListener("click", () => {
+  var requestBody = {
+    content: document.getElementById("message-input").value
```
Comment (jslopezgithub) [inline week-5/F-webchat/exercise-2.js]:
```suggestion
var input = document.querySelector("#message-input");
 var requestBody = { "content": value };
```

File: week-5/F-webchat/exercise-1.js [hunk 26]
```diff
@@ -33,5 +34,11 @@ When you open index.html in your browser, it should display the existing message
 
 */
 
+// Write your code here
```
Comment (jslopezgithub) [inline week-5/F-webchat/exercise-1.js]: Exercise-3.js is missing

File: week-1-practise/level-2/4-currency-conversion-2.js (hunk 4 above)
Comment (milleymilli) [reply week-1-practise/level-2/4-currency-conversion-2.js]: That's good way to do it,,, but mine doesn't mean it's wrong. Does it?

File: week-2-practise/level-1/5-indexOf.js (hunk 5 above)
Comment (milleymilli) [reply week-2-practise/level-1/5-indexOf.js]: Of course,,,,, even there is no h value in the given array,,,, I put it there when I was giving to some examples my class mates,,,,it's my bad 😉

File: week-1-practise/level-2/4-currency-conversion-2.js (hunk 4 above)
Comment (jslopezgithub) [reply week-1-practise/level-2/4-currency-conversion-2.js]: Yours is correct as well.

File: week-5/F-webchat/exercise-1.js (hunk 23 above)
Comment (milleymilli) [reply week-5/F-webchat/exercise-1.js]: i won't see the out-put if i make it response.json() on the message list ,,,,,it's gonna be full of this,,,,,,,,[object Object],[object Object]

File: week-5/F-webchat/exercise-1.js (hunk 24 above)
Comment (milleymilli) [reply week-5/F-webchat/exercise-1.js]: mes.forEach(element => {
  element[content];
});
U MEAN  LIKE THIS,,,,,

File: week-5/F-webchat/exercise-1.js (hunk 23 above)
Comment (jslopezgithub) [reply week-5/F-webchat/exercise-1.js]: You need to parse the object array.

File: week-5/F-webchat/exercise-1.js (hunk 24 above)
Comment (jslopezgithub) [reply week-5/F-webchat/exercise-1.js]:
```suggestion
messages.forEach(message => {
            var paragraph = document.createElement("p");
            paragraph.appendChild(document.createTextNode(message.datetime + ": " + message.content));
            messageList.appendChild(paragraph);
        });
```

File: week-5/F-webchat/exercise-2.js (hunk 25 above)
Comment (milleymilli) [reply week-5/F-webchat/exercise-2.js]: value is undefiend in your request body ,,,,i guess .
if like this one is possible i think =>
  var input = document.querySelector("#message-input").value;
 var requestBody = { "content": value };

File: week-5/F-webchat/exercise-1.js (hunk 26 above)
Comment (milleymilli) [reply week-5/F-webchat/exercise-1.js]: it's in exercise-1 ,,,,, coz it says we have to add the setInterval to exercise-1

**Pull request closed without merging**